### PR TITLE
Mac: Turn off automatic GPU switching

### PR DESCRIPTION
**Pull request by atdotde**
We are not OpenGL heavy so this saves battery life.

Signed-off-by: Robert C. Helling <helling@atdotde.de>

This should address #1719 according to to https://codereview.qt-project.org/#/c/73839/

Please test (all my Macs have only a single GPU, so I cannot)

### Describe the pull request:

- [ ] Bug fix
- [x] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### CODE REVIEW

**Comment by janmulder** [comment]
While the fix looks sane, I'm not sure here (disclaimer: I'm not an Apple user :-). Are we switching this off for **all** Apple users by default, where, at the same time, any affected user, can put this line in his/hers info.plist. Something we can easily put in the documentation.

File: packaging/macosx/Info.plist.in [hunk 1]
```diff
@@ -18,5 +18,6 @@
 	<string>6.0</string>
 	<key>CFBundleVersion</key>
 	<string>@SHORT_VERSION@</string>
+	<key>NSSupportsAutomaticGraphicsSwitching</key><true/>
```
Comment (dirkhh) [inline packaging/macosx/Info.plist.in:21]: the commit title and commit content seem to disagree. I think you are turning automatic GPU switching **on**, not **off**

**Comment by dirkhh** [comment]
@janmulder no, this is for this specific app, not for the OS

**Comment by atdotde** [comment]
@janmulder I think it does, but my expectation would be that it doesn't matter since we use the GPU only for animation effects and the map (and not for a first person shooter). We don't turn off the GPU, just use the internal one.

@dirkhh You might be right, but i copy&pasted the line from the link I gave above (that might be wrong). Will check the documentation again.

In any case, I cannot test this since I don't have a dual GPU Mac. Where is the result of the Travis build so @justmeMuc (the original poster) can test?

**Comment by atdotde** [comment]
@dirkhh see last comment on https://stackoverflow.com/questions/12445561/nssupportsautomaticgraphicsswitching-support

**Comment by dirkhh** [comment]
Or see this: http://supermegaultragroovy.com/2016/12/10/auto-graphics-switching/
All I'm saying is that  this is actually turning switching ON, not OFF. And that's the whole point. Because whenever we create an OpenGL context, apparently Apple thinks we need the external GPU. But with this flag set, it will switch us to the internal GPU if we don't burn a lot of cycles (which we don't)

**Comment by dirkhh** [comment]
Childish angry comment deleted.
I'll merge this manually and adjust the commit message to reflect what the code does.

**Comment by atdotde** [comment]
Ah, sorry. I did not realise (read properly) you talked about the commit message and not changing the option to false.

**Comment by dirkhh** [comment]
This is one of the things that I **really** dislike about GitHub's UI. You cannot comment "in context" on commit messages, only on the code itself.
Anyway, resolved now.

**Pull request closed without merging**